Realm/RLMSyncConfiguration.h

```diff
@@ -84,25 +138,31 @@ typedef void(^RLMClientResetAfterBlock)(RLMRealm * _Nonnull beforeFrozen, RLMRea
 
 /**
  An enum which determines file recovery behvaior in the event of a client reset.
- - note: Defaults to `RLMClientResetModeManual`
+ @note: Defaults to `RLMClientResetModeRecoverUnsycnedChanges`
 
- - see: `RLMClientResetMode`
- - see: https://docs.mongodb.com/realm/sync/error-handling/client-resets/
+ @see: `RLMClientResetMode`
+ @see: https://docs.mongodb.com/realm/sync/error-handling/client-resets/
 */
 @property (nonatomic) RLMClientResetMode clientResetMode;
 
 /**
  A callback which notifies prior to  prior to a client reset occurring.
- - see: `RLMClientResetBeforeBlock`
+ @see: `RLMClientResetBeforeBlock`
  */
 @property (nonatomic, nullable) RLMClientResetBeforeBlock beforeClientReset;
 
 /**
  A callback which notifies after a client reset has occurred.
- -see: `RLMClientResetAfterBlock`
+ @see: `RLMClientResetAfterBlock`
  */
 @property (nonatomic, nullable) RLMClientResetAfterBlock afterClientReset;
 
+/**
+ 
+ */
+@property (nonatomic, nullable) RLMSyncErrorReportingBlock manualClientResetHandler;
```

Comment:
I think you're right. The idea is that all the other SDKs are trying to make the manual mode handler set similar to the other three modes. It'd be odd to have four modes, but one of the modes would have a handler set elsewhere. But having this manualClientResetHandler kind of makes it uglier. I suppose we do need this at least privately so that `setErrorHandler` can invoke the callback passed down from the swift layer.